crates/nu-parser/tests/test_parser.rs

```diff
@@ -781,6 +781,19 @@ fn test_nothing_comparison_eq() {
         )
     ))
 }
+#[rstest]
+#[case(b"let a = 1 err> /dev/null")]
+#[case(b"let a = 1 out> /dev/null")]
+#[case(b"let a = 1 out+err> /dev/null")]
+#[case(b"mut a = 1 err> /dev/null")]
+#[case(b"mut a = 1 out> /dev/null")]
+#[case(b"mut a = 1 out+err> /dev/null")]
+fn test_redirection_with_letmut(#[case] phase: &[u8]) {
```

Comment:
do you think we could have, in addition to that, a little test that makes sure running these commands give the appropriate errors?

you know the
```rust
let actual = nu!("let a = 1 err> /dev/null");
assert!(actual.err.contains("Caused by this"));
```
type of tests :yum:

Reply:
Added a new error in `nu-protocol/src/parser_error.rs` to produce better error messages and now tests will check `working_set.parse_errors` contains expected error.

<img width="440" alt="err"  src="https://github.com/nushell/nushell/assets/1991933/b0980303-38b6-4280-ae93-1eae35cb85a6">